cores/arduino/UARTClass.cpp

```diff
@@ -183,8 +183,7 @@ size_t UARTClass::write( const uint8_t uc_data )
   {
     // If busy we buffer
     int l = (_tx_buffer->_iHead + 1) % UART_BUFFER_SIZE;
-    while (_tx_buffer->_iTail == l)
-      ; // Spin locks if we're about to overwrite the buffer. This continues once the data is sent
+    while (*(volatile int*)&(_tx_buffer->_iTail) == l); // Spin locks if we're about to overwrite the buffer. This continues once the data is sent
```

Comment:
It looks like the RingBuffer class is only used by the UART driver, so I think we can just declare _iTail as volatile in RingBuffer.h, rather than forcing volatile semantics for each access.